File: tests/fixtures/golden/objc/sample.m
#import <Foundation/Foundation.h>

@protocol Identifiable <NSObject>
@property (nonatomic, copy) NSString *id;
@end

typedef NS_ENUM(NSInteger, Status) {
    StatusActive,
    StatusInactive
};

@interface BaseEntity : NSObject
@property (nonatomic, copy) NSString *name;
- (instancetype)initWithName:(NSString *)name;
@end

@implementation BaseEntity
- (instancetype)initWithName:(NSString *)name {
    self = [super init];
    if (self) {
        _name = [name copy];
    }
    return self;
}
@end

@interface User : BaseEntity <Identifiable>
@property (nonatomic, copy) NSString *id;
@property (nonatomic, assign) Status status;
- (instancetype)initWithId:(NSString *)id name:(NSString *)name;
- (NSString *)process;
@end

@implementation User
- (instancetype)initWithId:(NSString *)id name:(NSString *)name {
    self = [super initWithName:name];
    if (self) {
        _id = [id copy];
        _status = StatusActive;
    }
    return self;
}

- (NSString *)process {
    return [self helperCall:self.name];
}

- (NSString *)helperCall:(NSString *)input {
    return [NSString stringWithFormat:@"processed_%@", input];
}
@end

File: tests/fixtures/golden/go/sample.go
package sample

import (
	"context"
	"fmt"
)

// Identifiable is implemented by anything with an ID.
type Identifiable interface {
	GetID() string
}

// Status enumerates active states.
type Status struct {
	Label string
}

// BaseEntity is the base type.
type BaseEntity struct {
	name string
}

// User extends BaseEntity and implements Identifiable.
type User struct {
	BaseEntity
	id     string
	status Status
}

// GetID satisfies Identifiable.
func (u *User) GetID() string {
	return u.id
}

// Process returns a processed label.
func (u *User) Process(ctx context.Context) string {
	return u.helperCall(ctx, u.name)
}

// helperCall is a private helper.
func (u *User) helperCall(ctx context.Context, input string) string {
	fmt.Println(input)
	return "processed_" + input
}

// Build is a package-level function.
func Build(id string) *User {
	return &User{id: id}
}
